src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -5398,22 +5398,21 @@ void MacroAssembler::multiply_to_len(Register x, Register xlen, Register y, Regi
 // Count bits of trailing zero chars from lsb to msb until first non-zero
 // char seen. For the LL case, shift 8 bits once as there is only one byte
 // per each char. For other cases, shift 16 bits once.
-void MacroAssembler::ctzc_bit(Register Rd, Register Rs, bool isLL,
-                              Register tmp1, Register tmp2) {
+void MacroAssembler::ctzc_bits(Register Rd, Register Rs, bool isLL,
+                               Register tmp1, Register tmp2) {
+  int step = isLL ? 8 : 16;
   if (UseZbb) {
-    assert_different_registers(Rd, Rs, tmp1);
-    int step = isLL ? 8 : 16;
+    assert_different_registers(Rd, tmp1);
     ctz(Rd, Rs);
     andi(tmp1, Rd, step - 1);
     sub(Rd, Rd, tmp1);
```

Comment:
Seems these 2 lines can be replaced with single instruction:
```
andi Rd, Rd, -step
```

Reply:
Great! I have fixed accordingly and remove the first assertion. Thanks.